CONTRIBUTING.md

```diff
@@ -32,6 +32,26 @@ channel on Freenode](http://webchat.freenode.net/?channels=grpc), or send an
 email to [grpc-io@googlegroups.com](grpc-io@googlegroups.com) to discuss
 beforehand.
 
+## Pull Requests & Commits
+
+We have few conventions for keeping history clean and making code reviews easier
+for reviewers:
+
+* First line of commit messages should be in format of
+
+  `package-name: summary of change`
+
+  for example:
+
+  `core: fix infinite loop`
```

Comment:
I wanted to make it more descriptive but honestly wasn't sure what are the current conventions and what are the ones that we want so I didn't want to jump the gun cause technically I'm not even an maintainer ;)

Updated. PTAL.